src/app/portlets/dot-edit-page/components/dot-edit-page-info/dot-edit-page-info.component.spec.ts

```diff
@@ -0,0 +1,159 @@
+            const pageTitleEl: HTMLElement = de.query(By.css('.page-info__title')).nativeElement;
+            expect(pageTitleEl.textContent).toContain('A title');
+        });
+
+        it('should set page url', () => {
+            const pageUrlEl: HTMLElement = de.query(By.css('.page-info__url')).nativeElement;
+            expect(pageUrlEl.textContent).toEqual('an/url/test');
+        });
+
+        it('should not have error messages', () => {
+            const lockedByMessage: DebugElement = de.query(By.css('.page-info__locked-by-message'));
+            const cantEditMessage: DebugElement = de.query(By.css('.page-info__cant-edit-message'));
+
+            expect(lockedByMessage === null && cantEditMessage === null).toBe(true);
+        });
+
+        it('should copy to clipboard url', () => {
+            spyOn(dotGlobalMessageService, 'display');
+            spyOn(component, 'copyUrlToClipboard').and.callThrough();
+            spyOn(document, 'execCommand').and.returnValue(true);
+
+            const copyUrlButton: DebugElement = de.query(By.css('.page-info__copy-url'));
+
+            copyUrlButton.nativeElement.click();
+
+            expect(component.copyUrlToClipboard).toHaveBeenCalledTimes(1);
+            expect(document.execCommand).toHaveBeenCalledWith('copy');
+            expect(dotGlobalMessageService.display).toHaveBeenCalledWith('Copied to clipboard');
+        });
+    });
+
+    describe('locked messages', () => {
+        describe('locked by another user', () => {
+            let lockedMessage: DebugElement;
+
+            beforeEach(() => {
+                component.pageState = new DotRenderedPageState(
+                    mockUser,
+                    JSON.parse(
+                        JSON.stringify({
+                            ...mockDotRenderedPage,
+                            page: {
+                                ...mockDotRenderedPage.page,
+                                canEdit: true,
+                                lockedBy: 'another-user'
+                            }
+                        })
+                    )
+                );
+                fixture.detectChanges();
+                lockedMessage = de.query(By.css('.page-info__locked-by-message'));
+            });
+
+            it('should have message', () => {
+                expect(lockedMessage.nativeElement.textContent.trim()).toEqual('Page is locked by...');
+            });
+
+            it('should blink', fakeAsync(() => {
+                spyOn(lockedMessage.nativeElement.classList, 'add');
+                spyOn(lockedMessage.nativeElement.classList, 'remove');
+                component.blinkLockMessage();
+
+                expect(lockedMessage.nativeElement.classList.add).toHaveBeenCalledWith('page-info__locked-by-message--blink');
+                tick(500);
+                expect(lockedMessage.nativeElement.classList.remove).toHaveBeenCalledWith('page-info__locked-by-message--blink');
+            }));
```

Comment:
I think this test should call the 'onLockerClick' method instead the blinkLockMessage

Reply:
Why? the blink can happen for more reasons that just the locker being click.